state/executor.go

```diff
@@ -749,15 +780,24 @@ func (t *Transition) applyCreate(c *runtime.Contract, host runtime.Host) *runtim
 	}()
 
 	// check if contract creation allow list is enabled
-	if t.deploymentAllowlist != nil {
-		role := t.deploymentAllowlist.GetRole(c.Caller)
+	if t.deploymentAllowList != nil {
+		role := t.deploymentAllowList.GetRole(c.Caller)
 
 		if !role.Enabled() {
 			return &runtime.ExecutionResult{
 				GasLeft: 0,
 				Err:     runtime.ErrNotAuth,
 			}
 		}
+	} else if t.deploymentBlockList != nil {
+		role := t.deploymentBlockList.GetRole(c.Caller)
+
+		if role == addresslist.EnabledRole {
```

Comment:
Check `AdminRole` as well?

```suggestion
		if role.Enabled() {
```

Reply:
https://github.com/0xPolygon/polygon-edge/pull/1388#discussion_r1171011573